mlos_bench/mlos_bench/environments/__init__.py

```diff
@@ -86,10 +362,22 @@
 used in different settings (e.g., local machine, SSH accessible machine, Azure VM,
 etc.) without having to change the Environment config.
 
-Variable Propagation
-++++++++++++++++++++
-TODO: Document how variable propagation works in the script environments using
-required_args, const_args, etc.
+Variable propagation rules described in the previous section for the environment
+configs also apply to the :py:mod:`Service <mlos_bench.services>`
+configurations.
+
+That is, every parameter defined in the Service config can be overridden by a
+corresponding parameter from the global config or the command line.
+
+All global configs, command line parameters, Environment ``const_args`` and
+``required_args`` sections, and Service config parameters thus form one flat
+name space of parameters.  This imposes a certain risk of name clashes, but also
```

Comment:
At some point we may want to define some "best practices" for variable naming to avoid clashes and make things more reusable.

Reply:
totally agree!

Reply:
Then we can close these, merge this, and tackle them later.

Reply:
> At some point we may want to define some "best practices" for variable naming to avoid clashes and make things more reusable.

https://github.com/microsoft/MLOS/issues/963